src/app/students/(hooks)/useStudents.tsx

```diff
@@ -1,38 +1,50 @@
 'use client'
-import { useState, createContext, useContext, ChangeEvent, PropsWithChildren, useMemo } from 'react'
+import { useState, createContext, useContext, PropsWithChildren, useMemo } from 'react'
 import { Student } from '@/_temp_types/student'
 
-type DropdownOption  = {
-  label: string;
-  value: string;
+type DropdownOption = {
+    label: string;
+    value: string;
 }
 
 type StudentsContextType = {
-  displayStudents: Student[];
-  currentSections: DropdownOption[];
+    displayStudents: Student[];
+    currentSections: DropdownOption[];
+    fetchStudents: () => Promise<void>;
 }
-
 const StudentsContext = createContext<StudentsContextType | undefined>(undefined)
 
 const useStudentsProvider = (): StudentsContextType => {
-    const [studentsParse, setStudentsParse] = useState<Student[]>([])
     const [displayStudents, setDisplayStudents] = useState<Student[]>([])
     const [sections, setSections] = useState<DropdownOption[]>([])
+    const fetchStudents = async () => {
+        const fixedCourseNum = 1  // TODO: replace with actual course number
+        const baseURL = process.env.NEXT_PUBLIC_DB_HOST
```

Comment:
didnt you change this name?

Reply:
yes, it's changed on the newer PRs. This PR was meant to be starting point for the pagination PR but since I have just merged that, I will close this PR.